csharp/ql/test/library-tests/csharp9/record.ql

```diff
@@ -0,0 +1,11 @@
+import csharp
+
+query predicate types(Class t, string i) {
```

Comment:
Do we need a way to distinguish `class` from `record`? Should we add an extra predicate to `Class` which would hold if the `class` was extracted from a `record` declaration?

Reply:
Yes, I think we should have a `record_class(unique int id: @class_type ref)` relation, and then have

```ql
class Record extends Class {
  Record() { record_class(this) }
}
```

Reply:
@hvitved I suggest doing this later. Roslyn 3.9 seems to have the `ITypeSymbol.IsRecord` property that we could use during the extraction. Currently there seems to be no easy way to distinguish a `class` or a `record` symbol.

Reply:
Sorry, coming back on this. I pushed an extra commit with some QL changes that handles both CIL and C# extraction. The above Roslyn API seems to do almost the same: [IsRecord](https://github.com/dotnet/roslyn/blob/48b6c31d3112bd4fd23a256fe3dfc205177a1c79/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L559-L566), and the underlying [FindValidCloneMethod](https://github.com/dotnet/roslyn/blob/48b6c31d3112bd4fd23a256fe3dfc205177a1c79/src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordClone.cs#L145).